.travis.yml

```diff
@@ -1,14 +1,27 @@
-# GCE VMs have better performance (will be upgrading to premium VMs soon)
-sudo: required
+dist: xenial
+group: stable
 
 language: scala
-jdk: openjdk8
 
 cache:
   directories:
     - $HOME/.ivy2/cache
     - $HOME/.sbt
     - $HOME/.rvm/
+    - $HOME/.sdkman
+
+before_install:
+  # adding $HOME/.sdkman to cache would create an empty directory, which interferes with the initial installation
+  - "[[ -d /home/travis/.sdkman/ ]] && [[ -d /home/travis/.sdkman/bin/ ]] || rm -rf /home/travis/.sdkman/"
+  - curl -sL https://get.sdkman.io | bash
+  - echo sdkman_auto_answer=true > /home/travis/.sdkman/etc/config
+  - source "/home/travis/.sdkman/bin/sdkman-init.sh"
+
+install:
+  - sdk install java $(sdk list java | grep -o "$ADOPTOPENJDK\.[0-9\.]*hs-adpt" | tail -1)
```

Comment:
I'm not so sure about this implementation as it seems `sdk list` wasn't built to be scripted around.

Particularly the `tail -1` it seems to me to be there to deal with the `sdk install java 11.0.3.hs-adpt` example at the bottom:

```
08:17:43 $ sdk list java
================================================================================
Available Java Versions
================================================================================
 Vendor        | Use | Version      | Dist    | Status     | Identifier
--------------------------------------------------------------------------------
 AdoptOpenJDK  |     | 12.0.1.j9    | adpt    |            | 12.0.1.j9-adpt
               |     | 12.0.1.hs    | adpt    |            | 12.0.1.hs-adpt
               |     | 11.0.3.j9    | adpt    |            | 11.0.3.j9-adpt
               |     | 11.0.3.hs    | adpt    |            | 11.0.3.hs-adpt
               |     | 8.0.212.j9   | adpt    |            | 8.0.212.j9-adpt
               |     | 8.0.212.hs   | adpt    |            | 8.0.212.hs-adpt
...
================================================================================
Use the Identifier for installation:

    $ sdk install java 11.0.3.hs-adpt
================================================================================
```

It looks like SDKMAN! only tracks the latest patch version of major versions, so if anything `head -1` is probably better, to get the latest in the list, not the version used as an example.

```suggestion
  - sdk install java $(sdk list java | grep -o "$ADOPTOPENJDK\.[0-9\.]*hs-adpt" | head -1)
```

Alternatively should we use SDKMAN!'s Candidates Service (https://github.com/sdkman/sdkman-candidates) API?   Assuming that's a reliably stable API we could use:

```suggestion
  - sdk install java $(curl -Ls "$SDKMAN_CANDIDATES_API/candidates/java/$SDKMAN_PLATFORM/versions/all" | tr ',' '\n' | grep "$ADOPTOPENJDK\.[0-9\.]*\.hs-adpt")
```

which behaves like this:

```bash
08:29:29 $ for ADOPTOPENJDK in 8 11; do
> curl -Ls "$SDKMAN_CANDIDATES_API/candidates/java/$SDKMAN_PLATFORM/versions/all" | tr ',' '\n' | grep "$ADOPTOPENJDK\.[0-9\.]*\.hs-adpt"
> done
8.0.212.hs-adpt
11.0.3.hs-adpt
```

Reply:
I reached out to the SDKMAN! project, which prefers that initial points of contact happen on their Slack, so the link to my thread on their Slack is https://sdkman.slack.com/archives/CJK9DMV6V/p1562658566004100.

Reply:
here's using head - https://github.com/scala/scala/pull/7931/commits/5f04df40618f9213e36be38d687a7ba550007afd